pkg/certifier/certify/certify_test.go

```diff
@@ -42,23 +45,43 @@ func newMockQuery() certifier.QueryComponents {
 }
 
 // GetComponents returns components for test
-func (q *mockQuery) GetComponents(ctx context.Context, compChan chan<- *certifier.Component) error {
+func (q *mockQuery) GetComponents(ctx context.Context, compChan chan<- interface{}) error {
 	compChan <- testdata.RootComponent
 	return nil
 }
 
+type mockUnknownQuery struct {
+}
+
+// NewMockQuery initializes the mockQuery to query for tests
+func newMockUnknownQuery() certifier.QueryComponents {
+	return &mockUnknownQuery{}
+}
+
+// GetComponents returns components for test
+func (q *mockUnknownQuery) GetComponents(ctx context.Context, compChan chan<- interface{}) error {
+	compChan <- assembler.AttestationNode{}
+	return nil
+}
+
 func TestCertify(t *testing.T) {
 	ctx := logging.WithLogger(context.Background())
 
+	err := RegisterCertifier(osv.NewOSVCertificationParser, certifier.CertifierOSV)
+	if err != nil && !strings.Contains(err.Error(), "the certifier is being overwritten") {
```

Comment:
Could the source return a specific error? Comparing strings in errors is not the ideal implementation.

Reply:
Declare a specific error `var e = errors.New("error")` at the package level and check for that specific error with `errors.Is`. How about something like this https://gosamples.dev/check-error-type/?

Reply:
nvm done.